### ISSUE TITLE
Update runbook/alerting for KafkaError _MAX_POLL_EXCEEDED

### ISSUE THREAD

**Issue by robrap** [edx/edx-arch-experiments#123]
To do list:
- [X] ~~Is there additional observability possible for dead consumers?~~
  - No. Kafka should just rebalance as needed, and for now, we don't seem to be left with consumers not running. 
- [x] Update runbook with `_MAX_POLL_EXCEEDED` details.
  - [x] Add general note about `OtherTransaction/Message/Kafka/Topic/Named/None`. 
- [x] Add observability for `fatal` and `retriable`. Logs, or just custom attributes? Add `has_message` attribute?
- [x] Determine how to handle KafkaError `fatal` and `retriable` appropriately. (Waiting on [support ticket](https://support.confluent.io/hc/en-us/requests/133348).)
  - [x] Assuming we raise some errors and kill loop, should we test that the process gets re-run? 

**Notes:**

We've seen a couple of the following errors in Production:
```
KafkaError{code=_MAX_POLL_EXCEEDED,val=-147,str=\"Application maximum poll interval (300000ms) exceeded by 221ms\"}
```
The setting `max.poll.interval.ms` (see https://docs.confluent.io/platform/current/installation/configuration/consumer-configs.html#max-poll-interval-ms) is currently using the default of 5 minutes.

The problem might be from an actual failing consumer, or getting hung on processing some message. The longest transaction trace captured wasn't long enough. It does look like there are salesforce service calls that could be slow.

* Here is NRQL for the errors: https://onenr.io/0bRKa42MpQE
* A link to the error trace: https://onenr.io/07j90AOXxRO

Open questions:
- Does the consumer recover? Is the consumer added back and does it process messages again, or is it just looping and getting no messages?
- Are any messages lost? It doesn't seem like they would be.
- Do we want to remove this from any of the alerts, or just note in the runbook that the consumers should recover? Should we note how to tell if all consumers are processing data?

**Comment by robrap** [comment]
Note: these result in a New Relic transaction name `OtherTransaction/Message/Kafka/Topic/Named/None`. Having the new custom attributes (separate ticket) would enable us to see the actual topic name.

**Comment by robrap** [comment]
There are transaction traces (e.g. https://onenr.io/0bRmN1Z6Yjy) where the call to `edx.my.salesforce.com` took 15 minutes. What is the ideal solution for making calls like this? Would we want it to fail faster, and be put on a retry queue? Other?

**Comment by robrap** [comment]
In the last 30 minutes, it looks like there are 6 IPs checking in, but one has flat-lined in regards to CPU usage, so I have a feeling it is a dud. How could we detect and restart?

**Comment by robrap** [comment]
According to Confluent Kafka, we have 3 consumers, each covering 2 partitions. See https://confluent.cloud/environments/env-57zr7z/clusters/lkc-6kvzrj/clients/consumer-lag/course_discovery_prod

We probably only need 2 or 3, but that doesn't answer earlier questions of detecting dead consumers (if we have any).

**Comment by robrap** [comment]
Does it seem reasonable that we aren't losing any messages, but we just need to determine if we are losing consumers and how to appropriately restart them?

**Comment by robrap** [comment]
I added the following comment to Slack here: https://twou.slack.com/archives/C03109K3V4J/p1669932386473219 (private link)

@chorvath]: I’ve got some error handling questions:
1. Can you explain what fatal vs retriable mean in https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#kafkaerror? Does fatal mean the infinite loop should be exited, because Kafka will no longer listen to the consumer? What errors could be both non-fatal and non-retriable, and what would that mean?
2. The example code in https://docs.confluent.io/kafka-clients/python/current/overview.html#python-code-examples shows _PARTITION_EOF. Is there anything special about this error, or is it just an example of an error that is not fatal, where you’d want to continue looping?

**Comment by robrap** [comment]
* Added custom attributes in https://github.com/edx/edx-arch-experiments/issues/121
* Created new ticket https://github.com/edx/edx-arch-experiments/issues/139
